### DISCUSSION TITLE
Dynamic sized watermark?

### DISCUSSION THREAD

**Discussion by 9mm** [libvips/libvips#3472] [Help wanted]
Hey,

I'm using ActiveStorage + Rails 7 to generate the following variant:

```
    file.variant :square_medium,
      format: 'jpg',
      resize_to_limit: [1280, 1280],
      saver: { subsample_mode: 'on', strip: true, quality: 85 },
      composite: ['app/assets/images/watermark_square_medium.png', {
        gravity: 'south-west',
        offset: [10, 10]
      }]   
```

The problem is I can't seem to dynamically size the watermark based on the width. I want it to be roughly 20%. of the width.

Is something like this possible so I don't need to make a watermark for every single variant size? That actually would be problematic too as I'm using `resize_to_limit` and it's not certain whether the image is smaller than the total width 🤔

**Comment by jcupitt** [comment]
Hi @9mm,

I suppose you could use a SVG watermark, would that work? You could load it at the right scale and then composite that on the image.

Alternatively you could make a large watermark and shrink that to a target size for compositing. Quality might not be as nice though.

(I've not used `image_processing` much, you might get better advice there)
